java/org/brotli/wrapper/enc/Encoder.java

```diff
@@ -22,18 +23,76 @@
   final ByteBuffer inputBuffer;
   boolean closed;
 
+  /**
+   * https://www.brotli.org/encode.html#aa6f
+   * See encode.h, typedef enum BrotliEncoderMode
+   */
+  public enum Mode {
+    /**
+     * Default compression mode.
+     * In this mode compressor does not know anything in advance about the properties of the input.
+     */
+    GENERIC(0),
+    /**
+     * Compression mode for UTF-8 formatted text input.
+     */
+    TEXT(1),
+    /**
+     * Compression mode used in WOFF 2.0.
+     */
+    FONT(2);
+
+    private final int value;
+
+    Mode(int value) {
+        this.value = value;
+    }
+
+    public int value() {
+        return value;
+    }
+
+    public static Mode of(int value) {
+        switch (value) {
+            case 0:
+                return GENERIC;
+            case 1:
+                return TEXT;
+            case 2:
+                return FONT;
+            default:
+                throw new IllegalArgumentException("Unknown mode: " + value);
+        }
+    }
+
+    public static Mode of(String name) {
+        switch (name.toLowerCase(Locale.ROOT)) {
+            case "generic":
+                return GENERIC;
+            case "text":
+                return TEXT;
+            case "font":
+                return FONT;
+            default:
+                throw new IllegalArgumentException("Unknown mode: " + name);
+        }
+    }
+  }
+
   /**
    * Brotli encoder settings.
    */
   public static final class Parameters {
     private int quality = -1;
     private int lgwin = -1;
+    private Mode mode = Mode.GENERIC;
```

Comment:
Prever to have "null" value, to represent default and does not override encoder setting in case default wasn't changed.

Reply:
Done